x-pack/plugins/infra/server/lib/alerting/log_threshold/log_threshold_executor.ts

```diff
@@ -106,6 +111,28 @@ export const createLogThresholdExecutor = (libs: InfraBackendLibs) =>
         },
       });
 
+      if (actions && actions.length > 0) {
+        const indexedStartedAt = getAlertStartedDate(id) ?? startedAt.toISOString();
+        const relativeViewInAppUrl = getLogsAppAlertUrl(new Date(indexedStartedAt).getTime());
+        const viewInAppUrl = basePath.publicBaseUrl
+          ? new URL(basePath.prepend(relativeViewInAppUrl), basePath.publicBaseUrl).toString()
+          : relativeViewInAppUrl;
+
+        const sharedContext = {
+          timestamp: new Date().toISOString(),
+          viewInAppUrl,
+        };
+        actions.forEach((actionSet) => {
+          const { actionGroup, context } = actionSet;
+          alert.scheduleActions(actionGroup, { ...sharedContext, ...context });
+        });
+      }
+
+      alert.replaceState({
+        alertState: AlertStates.ALERT,
```

Comment:
I moved the `AlertState.ALERT` here from the `alertFactory` function as it's always the same.